java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -337,3 +337,94 @@ JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_releaseNamesHandle
     }
     checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,names));
 }
+
```

Comment:
How do we ensure we're not leaking any memory?

Reply:
`releaseNamesHandle` isn't used anymore, the `OrtSession` stores the names in the Java heap. I can make another PR to remove it after this one has merged as otherwise it'll conflict with the session options one.

As to the metadata, all of that is extracted once when constructMetadata is called, and theres a matching free to each of the allocate calls in the C function. The metadata then is allocated on the Java heap, and cached in the `OrtSession`.